packages/flutter/lib/src/widgets/list_wheel_scroll_view.dart

```diff
@@ -0,0 +1,193 @@
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+
+import 'framework.dart';
+import 'scroll_controller.dart';
+import 'scroll_physics.dart';
+import 'scrollable.dart';
+
+/// A box in which children on a wheel can be scrolled.
+///
+/// This widget is similar to a [ListView] but with the restriction that all
+/// children must be the same size along the scrolling axis.
+///
+/// When the list is at the zero scroll offset, the first child is aligned with
+/// the middle of the viewport. When the list is at the final scroll offset,
+/// the last child is aligned with the middle of the viewport
+///
+/// The children are rendered as if rotating on a wheel instead of scrolling on
+/// a plane.
+class ListWheelScrollView extends StatelessWidget {
+  /// Creates a box in which children are scrolled on a wheel.
+  const ListWheelScrollView({
+    Key key,
+    this.controller,
+    this.physics,
+    this.diameterRatio: 2.0,
+    this.perspective: 0.003,
+    @required this.itemExtent,
+    this.clipToSize: true,
+    this.renderChildrenOutsideViewport: false,
+    @required this.children,
+  }) : assert(diameterRatio != null && diameterRatio > 0.0),
+       assert(
+         perspective != null && perspective >= 0.0 && perspective < 0.01,
+         'A perspective too high will be clipped in the z-axis and therefore '
+         'not renderable. Value must be between 0 and 0.01.'
+       ),
+       assert(itemExtent != null && itemExtent > 0.0),
+       assert(clipToSize != null),
+       assert(renderChildrenOutsideViewport != null),
+       assert(
+         !renderChildrenOutsideViewport || !clipToSize,
+         'Cannot renderChildrenOutsideViewport and clipToSize since children '
+         'rendered outside will be clipped anyway.'
+       ),
+       super(key: key);
+
+  /// An object that can be used to control the position to which this scroll
+  /// view is scrolled.
+  ///
+  /// A [ScrollController] serves several purposes. It can be used to control
+  /// the initial scroll position (see [ScrollController.initialScrollOffset]).
+  /// It can be used to control whether the scroll view should automatically
+  /// save and restore its scroll position in the [PageStorage] (see
+  /// [ScrollController.keepScrollOffset]). It can be used to read the current
+  /// scroll position (see [ScrollController.offset]), or change it (see
+  /// [ScrollController.animateTo]).
+  final ScrollController controller;
+
+  /// How the scroll view should respond to user input.
+  ///
+  /// For example, determines how the scroll view continues to animate after the
+  /// user stops dragging the scroll view.
+  ///
+  /// Defaults to matching platform conventions.
+  final ScrollPhysics physics;
+
+  /// {@macro flutter.rendering.wheelList.diameterRatio}
+  ///
+  /// Defaults to an arbitrary but aesthetically reasonable number of 2.0.
+  ///
+  /// Must not be null and must be positive.
```

Comment:
Why is "Must not be null and must be positive." not part of the template?

Reply:
this also applies to other docs here

Reply:
Moved everything into the templates.